server/lib/service/firestore.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:googleapis/firestore/v1.dart';
+import 'package:http/http.dart';
+
+import '../proto/rule.pb.dart';
+import '../utils.dart';
+
+/// A class that wraps around Cloud Firestore.
+class FirestoreClient {
+  /// The name of collection of users in Firestore.
+  static const usersCollectionName = 'users';
+
+  /// The name of the collection of rules in Firestore.
+  static const rulesCollectionName = 'rules';
+
+  /// The name of the encrypted refresh token field in Firestore.
+  static const encryptedRefreshTokenFieldName = 'encryptedRefreshToken';
+
+  /// The Firestore API.
+  final FirestoreApi _api;
+
+  /// The Google Cloud project ID.
+  final String projectId;
+
+  /// The Cloud Firestore database ID.
+  final String databaseId;
+
+  /// Creates an instance of [FirestoreClient].
+  FirestoreClient(
+      Client client, this.projectId, this.databaseId, String baseUrl)
+      : _api = FirestoreApi(client, rootUrl: baseUrl);
+
+  /// Adds a rule to the Firestore database.
+  ///
+  /// The [rule] is an instance of [Rule], a protobuf generated class.
+  ///
+  /// Throws an [ApiRequestError] if Firestore API returns an error.
+  Future<String> createRule(String userId, Rule rule) async {
+    final document = rule.toDocument();
+
+    return await _addDocument(
```

Comment:
`createRule()` returns the ruleId because the id for the rule document is generated by Firestore, whereas `createUser()` doesn't because we set its id to the parameter `userId`. 

Should I return the `userId` too? Or just make this clearer in the documentation?

Reply:
I have same question, if we specify userId in client side, what values you plan to use?

Reply:
I think it will be the F1 user id?  It is used in public api request, so user should know their F1 ids? But is it okay to save it in Firestore?

Reply:
The userId should be the `sub` field of the user's ID token, which is an obfuscated GAIA ID.

Reply:
Can you link the def of the userId token so we can find the `sub` field?

Reply:
Sure, here's the [link](https://developers.google.com/identity/protocols/oauth2/openid-connect#an-id-tokens-payload), and I'll add it to the documentation.